src/metabase/driver/sql_jdbc.clj

```diff
@@ -162,11 +164,15 @@
         (jdbc/execute! conn sql)))))
 
 (defmethod driver/add-columns! :sql-jdbc
-  [driver db-id table-name col->type]
-  (let [sql (first (sql/format {:alter-table (keyword table-name)
-                                :add-column (map (fn [[name type-spec]]
-                                                   (vec (cons name type-spec)))
-                                                 col->type)}
+  [driver db-id table-name column-definitions & {:keys [primary-key]}]
+  (mu/validate-throw [:maybe [:cat :keyword]] primary-key) ; we only support adding a single primary key column for now
+  (let [primary-key-column (first primary-key)
+        sql (first (sql/format {:alter-table (keyword table-name)
+                                :add-column (cond-> (map (fn [[name type-and-constraints]]
```

Comment:
Afraid my eyes might betray me again, but it doesn't look like this outer `cond->` has any transformation clauses.

Reply:
Good eyes! 71cc678749b10ef0216f4321ed8a35727f22c2be